stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -292,7 +292,8 @@ end
 # in A to matrices of type T and sizes given by n[k:end].  n is an array
 # so that the same promotion code can be used for hvcat.  We pass the type T
 # so that we can re-use this code for sparse-matrix hcat etcetera.
-promote_to_arrays_(n::Int, ::Type{Matrix}, J::UniformScaling{T}) where {T} = copyto!(Matrix{T}(undef, n,n), J)
+promote_to_arrays_(n::Int, ::Type, a::T) where {T<:Number} = copyto!(Vector{T}(undef, 1), a)
```

Comment:
A little unfortunate that for every scalar it will now allocate a `Vector` first … seems like it would be better to change the logic elsewhere.